File: Python/7py 1/easyline.py
"""
|--------------------------------------------------------------------------
| Easy Line
|--------------------------------------------------------------------------
|
"""

def easyline(n):
   return easyline(n-1) * (4*n-2) // n if n else 1


from math import factorial
def easyline(n):
   return factorial(2 * n) // factorial(n)**2


def easyline(n):
   row = [1]
   for i in range(1, n+1):
      row.append(row[-1] * (n-i+1) // i)
   return sum(i ** 2 for i in row)


print(easyline(7), 3432)
print(easyline(13), 10400600)
print(easyline(17), 2333606220)
print(easyline(19), 35345263800)

File: PHP/7php 1/easyLine.php
<?php
/*
|--------------------------------------------------------------------------
| Easy Line
|--------------------------------------------------------------------------
|
*/

function easyLine($n) {
   $res = 1;
   for ($i = 1; $i <= $n; $i++) $res *= ($n + $i) / $i;
   return round(log($res));
}


function easyline($n) {
   return round(array_sum(array_map('log', range($n+1, 2*$n))) - array_sum(array_map('log', range(1, $n))));
}


function easyline($n) {
   $sum = 0;
   for ($i = 1; $i <= $n; $i++) {
      $sum += log(($n + $i) / $i);
   }
   return round($sum);
}


print_r(easyLine(7) . ' ' . 8 . '
');
print_r(easyLine(13) . ' ' . 16 . '
');
print_r(easyLine(17) . ' ' . 22 . '
');
print_r(easyLine(19) . ' ' . 24 . '
');